examples/text-generation/utils.py

```diff
@@ -439,7 +444,10 @@ def setup_distributed_model(args, model_dtype, model_kwargs, logger):
 
     if load_to_meta:
         # Construct model with fake meta tensors, later will be replaced on devices during ds-inference ckpt load
-        with deepspeed.OnDevice(dtype=model_dtype, device="meta"):
+
+        deepspeed_device = "cpu" if keep_module_on_host else "meta"
```

Comment:
It seems in above code  line#438, if keep_module_on_host is True, the it will never come to this line, why we need this check again?